scripts/server.py

```diff
@@ -550,66 +567,544 @@ def handle_client(conn, addr):
+                            flush=True)
+                    except (AttributeError, ValueError, KeyError) as value_exec:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong accessing JSON data - {value_exec} not found',
+                            flush=True)
+                    except BaseException as dp_ex:
+                        print(
+                            f'BirdWeather Submission Error:: {retry_text}Detection POST - Something went wrong - {dp_ex}',
+                            flush=True)
+                    finally:
+                        if not can_retry:
+                            # Stop the loop as we can't be sure we can retry on this error
+                            break
+                        else:
+                            # We're retrying, retry after the calculated backoff time
+                            if request_backoff_time > 0:
+                                time.sleep(request_backoff_time)
+                                continue
+                else:
+                    print(
+                        f"BirdWeather Submission Error:: {retry_text}Detection POST did not succeed, detection_url, detection_json or soundscape_id missing or not supplied",
+                        flush=True)
+
+        else:
+            print("BirdWeather Submission Error:: Soundscape POST did not succeed, detection was not submitted",
+                  flush=True)
+
+    else:
+        print("BirdWeather Submission Error:: Soundscape POST - Soundscape URL or wav_data is missing or not supplied",
+              flush=True)
+
+
+def birdweather_submission_processor():
+    if debug_birdweather_submissions:
+        print("Starting: birdweather_submission_processor thread")
+
+    # Loop over the queue containing the data used for the BirdWeather submissions
+    while True:
+        if not bw_submission_queue.empty():
+            # Get the BirdWeather submission data, this is a dictionary containing the necessary data
+            bw_submission_data = bw_submission_queue.get()
+
+            # Verify we have data
+            if bw_submission_data is not None:
+                if debug_birdweather_submissions:
+                    print(
+                        f"Processing: Soundscape:{bw_submission_data['soundscape_url']}, Detection:{bw_submission_data['detection_url']}, Detection_Data:{bw_submission_data['detection_post_json']}")
+
+                # Perform the submission
+                birdweather_submit(bw_submission_data)
+            # Processing finished so the task is now done
+            bw_submission_queue.task_done()
+        # Sleep a bit between loops
+        time.sleep(1)
+
+
+def birdweather_soundscapes(mode, soundscape_filename_to_find, bw_soundscape_id_for_filename=None):
```

Comment:
Can you give this function a more meaningful name?

Reply:
changed to birdweather_soundscape_id_cache